### PR TITLE
Improve View Flagged Books admin verb

### PR DESCRIPTION
**Pull request by tigercat2000**
This changes the "View Flagged Books" admin verb into a datum/browser
UI. You can see all the books that have been flagged from here, review
them, and remove the flagged status or delete them.

Also refactors pencode into some __HELPER procs, and adds a
"reverse-pencode" proc. It's not perfect, due to the complexity of HTML
and my unwillingness to use BYONDregex, but it's good enough to get the
general idea of the formatting across. It's used for the new "view book"
panel.

### CODE REVIEW

**Comment by scrubmcnoob** [comment]
finally, i can now administrate where it matters the most and to my fullest potential.
thanks tiger

**Comment by DarkPyrolord** [comment]
Yes please, this will help loads, and also help remove the useless flags people put on books from time to time.

**Comment by Twinmold93** [comment]
No complaints with this. It'll make going through the flagged books SO much easier, and about time to remove the flagged ones that shouldn't be flagged from the list.

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -3019,6 +3019,59 @@
 		else
 			error_viewer.showTo(usr, null, href_list["viewruntime_linear"])
 
+	// Library stuff
+	else if(href_list["library_book_id"])
+		var/isbn = href_list["library_book_id"]
```
Comment (Crazylemon64) [inline code/modules/admin/topic.dm]: If an admin gets fed a bad topic datum, could they not be fed an arbitrary isbn, and thus possibly an unsanitized arbitrary query?

Comment (tigercat2000) [inline code/modules/admin/topic.dm]: Uhh... yes!
Any suggestions as to how to prevent this?

Comment (Crazylemon64) [inline code/modules/admin/topic.dm]: `var/isbn = sanitizeSQL(href_list["library_book_id")`
Would that not do it?

**Pull request merged**